lib/http/server.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const {
+  Promise,
+  RegExpPrototypeExec,
+  StringPrototypeEndsWith,
+  StringPrototypeIndexOf,
+  StringPrototypeLastIndexOf,
+  StringPrototypeReplace,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+  StringPrototypeToLocaleLowerCase,
+} = primordials;
+
+const { cwd, nextTick } = process;
+const console = require('internal/console/global');
+const { createReadStream } = require('fs');
+const { Server } = require('_http_server');
+const { URL, isURLInstance, pathToFileURL } = require('internal/url');
+const { sep } = require('path');
+const { emitExperimentalWarning } = require('internal/util');
+
+emitExperimentalWarning('http/server');
+
+if (module.isPreloading) {
+  const { parseArgs } = require('internal/util/parse_args/parse_args');
+  const { values: { port, host } } = parseArgs({ options: {
+    port: {
+      type: 'string',
+      short: 'p',
+    },
+    host: {
+      type: 'string',
+      short: 'h',
+    },
+  } });
+  nextTick(createSimpleServer, process.argv[1], port, host);
+  process.env.NODE_REPL_EXTERNAL_MODULE = 'node:http/server';
+  process.argv[1] = 'node:http/server';
+}
+
+function createSimpleServer(directory = cwd(), port = 0, host = 'localhost') {
+  const CWD_URL = isURLInstance(directory) ? directory : pathToFileURL(directory + sep);
+
+  const server = new Server(async (req, res) => {
+    const url = new URL(
+      req.url === '/' || StringPrototypeStartsWith(req.url, '/?') ?
+        './index.html' :
+        StringPrototypeReplace(new URL(req.url, 'root://').toString(), 'root://', '.'),
+      CWD_URL
+    );
```

Comment:
We should likely ensure that this filters out things like dotfiles (e.g. `/.foo` shouldn't be allowed)